src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion_img2img.py

```diff
@@ -241,19 +261,19 @@ def loop_body(step, args):
             params["scheduler"], num_inference_steps=num_inference_steps, shape=latents_shape
         )
 
-        t_start = self.get_timestep_start(num_inference_steps, strength, scheduler_state)
-        latent_timestep = scheduler_state.timesteps[t_start : t_start + 1].repeat(batch_size)
-        init_latents = self.scheduler.add_noise(init_latents, noise, latent_timestep)
-        latents = init_latents
+        latent_timestep = scheduler_state.timesteps[start_timestep : start_timestep + 1].repeat(batch_size)
+
+        latents = self.scheduler.add_noise(params["scheduler"], init_latents, noise, latent_timestep)
 
-        if debug:
+        # scale the initial noise by the standard deviation required by the scheduler
+        latents = latents * params["scheduler"].init_noise_sigma
+
+        if DEBUG:
             # run with python for loop
-            for i in range(t_start, len(scheduler_state.timesteps)):
+            for i in range(start_timestep, num_inference_steps):
```

Comment:
think it's safer to use `len(scheduler_state.timesteps)` that `num_inference_steps` because depending on the scheduler the could be some extra timesteps, for example if using `PNDMScheduler` with PRK stpes, it'll add some extra time timesteps

Reply:
:thinking: I was not aware of that. Thanks for letting me know. In that case it would also have to be changed [here](https://github.com/huggingface/diffusers/blob/df2b548e893ccb8a888467c2508756680df22821/src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py#L271).

Reply:
Hmm, it looks to me like [here](https://github.com/huggingface/diffusers/blob/25e82ec47c40d5619e0772ed68f4082da4bd6457/src/diffusers/schedulers/scheduling_pndm_flax.py#L200) it is making sure that the length is always equal to `num_inference_steps` by dropping some plms timesteps.

Or am I getting that wrong?

Reply:
Actually you are right, let's use `num_inference_steps` here. Because the timesteps are extended for 2nd order schedulers like Heun.

Reply:
Hmm, I'm not sure about this. Because the timesteps are extended for some schedulers, shouldn't we loop through the `timesteps` instead?

Reply:
Those schedulers are not part of flax yet no? (specifically the `HeunScheduler`),

Reply:
No, they are not.